test/gtest/uct/test_amo_and_or.cc

```diff
@@ -0,0 +1,94 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2001-2018.  ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#include "test_amo.h"
+
+
+class uct_amo_and_or_test : public uct_amo_test {
+public:
+
+    template <typename T>
+    void test_op(send_func_t send, T (*op)(T, T), T (*val)(unsigned)) {
+        /*
+         * Method: Add may random values from multiple workers running at the same
+         * time. We expect the final result to be the and/or of all these values.
+         * This is simplified version of add/xor test: operated value is costant
+         * for every worker to eliminate result to 0 or MAX_INT
+         */
+
+        mapped_buffer recvbuf(sizeof(T), 0, receiver());
+
+        T value = 0x0ff0f00f;
+        *(T*)recvbuf.ptr() = value;
+
+        T exp_result = value;
+        std::vector<uint64_t> op_vec;
+        for (unsigned i = 0; i < num_senders(); ++i) {
+             value = val(i);
+             op_vec.push_back(value);
+
+             for (unsigned j = 0; j < count(); ++j) {
+                 exp_result = op(exp_result, value);
+             }
+        }
+
+        run_workers(send, recvbuf, op_vec, false);
+
+        wait_for_remote();
+        EXPECT_EQ(exp_result, *(T*)recvbuf.ptr());
+    }
+};
+
+template <typename T>
+T and_op(T v1, T v2)
```

Comment:
maybe define and_op,or_op,... inside the class as static functions so they will not be part of global namespace

Reply:
moved into class